src/Java.Interop/Java.Interop/JniRuntime.JniValueManager.cs

```diff
@@ -590,6 +590,18 @@ public override void DestroyGenericArgumentState (IJavaPeerable value, ref JniVa
 		}
 
 		public override Expression CreateParameterFromManagedExpression (JniValueMarshalerContext context, ParameterExpression sourceValue, ParameterAttributes synchronize)
+		{
+			var r = CreateIntermediaryExpressionFromManagedExpression (context, sourceValue);
+			context.CleanupStatements.Add (DisposeObjectReference (r));
```

Comment:
In another "What was I thinking?" moment...  Is `CreateParameterFromManagedExpression()` even used from anywhere?

*Should it even exist*?

This method would be called when creating an `Expression` to convert a *managed* (C#) *parameter* to a Java parameter.

*What does that even mean*?  At least in the context of `jnimarshalmethod-gen.exe`, we never convert parameters *from* C# *to* Java.  We convert Java parameters to C#.  We convert C# *return values* to Java (hence `CreateReturnValueFromManagedExpression()`).  But parameters?

Which brings us to this line: As per our earlier offline discussion, we should *never* call `JniObjectReference.Dispose(ref JniObjectReference)`.  Meaning this line -- the line I'm commenting on -- should not exist at all; it's presence is a bug:

```csharp
context.CleanupStatements.Add (DisposeObjectReference (r));
```

That said...note that we don't see any use of `JniObjectReference.Dispose()` left in this PR.  Meaning this `CreateParameterFromManagedExpression()` *is not actually used*, at least not in any of our existing unit tests!

Is this method (now) dead code, as you've changed `CreateReturnValueFromManagedExpression()` to no longer call `CreateParameterFromManagedExpression()`?

I suspect you could replace this `CreateParameterFromManagedExpression()` method with `throw new NotImplementedException()` and nothing would break.

---

In any event, we shouldn't be calling `DisposeObjectReference()` here.  (There probably shouldn't even be a "here" here...)